### DISCUSSION TITLE
Can PC2 run in parallel or shorter time?

### DISCUSSION THREAD

**Discussion by superxii** [filecoin-project/lotus#9344] [General Q&A]
I am the beginner in lotus. I have pldege 4 sectors at a single time and PC 1 took around 8 Hrs for those 4 secotrs. And PC2 takes 1 hr for each pledge. But they are running PC2 synchronous, they run one by one. Is it possible that I can run the PC2 for the sectors together?

And I am not sure whether the time used is normal for PC1 with 4 sectors at the same time ?
There is no another tasks is running.

My Setup:
One Miner
One Worker

CPU: Intel E5 2678V3
GPU: NVIDIA 3090
RAM: ~1000GB

Worker handle PC1, PC2, C .

One more question
It is normal that the GPU is not always highly loaded for PC2/C2? I saw that the GPU is not always having a high GPU-Util when the PC2/C2 is running.
![image](https://user-images.githubusercontent.com/63416489/191438899-846f04a9-08d9-4f17-aa4d-6bab3bba0a28.png)

![image](https://user-images.githubusercontent.com/63416489/191437123-c422b565-a39a-4e3c-8e41-16854fb315d2.png)

**Comment by rjan90** [comment]
Hey @superxii! What kind of sectors sizes are you sealing? Your sealing times are most likely heavily impacted by your CPU which does not have the SHA-extensions.

For PC1 it´s normal to see sealing times around 3-5 hours with a good CPU and decent disks. For PC2 your should definitely see sealing times around 10-30 minutes depending on GPU. On followup question here is if you have installed CUDA? And if so, if you built lotus with the `FFI_USE_CUDA=1` enviroment variable https://lotus.filecoin.io/storage-providers/setup/prerequisites/#install-cuda. PC2 is also very read heavy, so if the sealing medium is on slow disks, that might slow down the sealing speed of PC2.

You can read more about the different tasks here: https://lotus.filecoin.io/storage-providers/get-started/tasks/

**Reply by superxii** [reply]
The sector size is in 32GB. Yea, I have installed with CUDA and set it into the env variable. I guess it should be some CPU problem/wrong setting on worker. Since if I run worker on miner machine, PC2 finished at 1X minutes.

**Reply by superxii** [reply]
But PC2/C2 can run in parallel? Just want to make sure it is not the nature from lotus

**Reply by rjan90** [reply]
They do not run in parallel nativel. But there is a tweak where you run multiple `lotus-workers` against a single GPU. As long as you have enough VRAM / RAM / Disk I/O, you should be able to run multiple PC2 / C2 at a time on a single server/machine.

**Reply by superxii** [reply]
![image](https://user-images.githubusercontent.com/63416489/191792442-fe25bf27-68d5-4115-8541-f8e12ac0c743.png)
![image](https://user-images.githubusercontent.com/63416489/191792628-ceebf6dc-1611-4215-abaf-0706689a849e.png)

Looks like something is wrong with my worker.Lotus shows my CPU in worker are working. 
But when I check the CPU usage with htop in workers, it didn't exactly the same with lotus-miner sealing workers.
I have stress tested with my CPU in worker, it works fine.

Also the GPU are not working even though worker got assign with onc PC2

Am I missing something on Lotus which caused CPU and GPU not to function well?

**Reply by superxii** [reply]
I may have found a root cause for slow process.
![image](https://user-images.githubusercontent.com/63416489/192097532-1645f0d6-81be-461a-adf9-36d1055cc573.png)
Is this related to a disk issue? But I have tested the connection between the worker to storage.It looks good, at least the speed is similar for miner to storage
